### PR TITLE
Call all Ids `id`, not `idx`

### PR DESCRIPTION
**Pull request by domenukk**

### CODE REVIEW

File: libafl_bolts/src/lib.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // For `std::simd`
 #![cfg_attr(nightly, feature(portable_simd))]
 // For `core::error`
+#![cfg_attr(nightly, feature(error_in_core))]
```
Comment (tokatoka) [inline libafl_bolts/src/lib.rs:15]: this is not needed. if you update to latest rust, it gives warnings

File: libafl/src/schedulers/accounting.rs [hunk 2]
```diff
@@ -266,15 +266,9 @@ where
...
-        state
-            .corpus()
-            .get(idx)?
-            .borrow_mut()
-            .metadata_map_mut()
-            .insert(AccountingIndexesMetadata::with_tcref(
-                indexes,
-                new_favoreds.len() as isize,
-            ));
+        state.current_testcase_mut()?.metadata_map_mut().insert(
```
Comment (tokatoka) [inline libafl/src/schedulers/accounting.rs:269]: this part is wrong?
it's not equivalent as before.

File: libafl/src/inputs/mod.rs [hunk 3]
```diff
@@ -53,7 +55,7 @@ pub trait Input: Clone + Serialize + serde::de::DeserializeOwned + Debug {
     }
 
     /// Generate a name for this input
-    fn generate_name(&self, idx: usize) -> String;
+    fn generate_name(&self, id: Option<CorpusId>) -> String;
```
Comment (tokatoka) [inline libafl/src/inputs/mod.rs:58]: why this guy is wrapped in Option?

Comment (domenukk) [reply libafl/src/inputs/mod.rs:58]: Because we don't set it in many places / just pass 0

File: libafl/src/events/mod.rs [hunk 4]
```diff
@@ -380,7 +380,7 @@ where
     fn name_detailed(&self) -> String {
         match self {
             Event::NewTestcase { input, .. } => {
-                format!("Testcase {}", input.generate_name(0))
+                format!("Testcase {}", input.generate_name(None))
```
Comment (domenukk) [inline libafl/src/events/mod.rs:383]: @tokatoka this is an example where we never passed a valid id

File: libafl/src/schedulers/accounting.rs (hunk 2 above)
Comment (domenukk) [reply libafl/src/schedulers/accounting.rs:269]: Aah thanks!

File: libafl_bolts/src/lib.rs (hunk 1 above)
Comment (domenukk) [reply libafl_bolts/src/lib.rs:15]: Yes but if you don't update it it doesn't compile, maybe remove it in a few weeks?

**Pull request merged**